### ISSUE TITLE
What *is* the default for `guess_lang`?

### ISSUE THREAD

**Issue by oprypin** [facelessuser/pymdown-extensions#1190]
1. https://github.com/facelessuser/pymdown-extensions/blob/6c3bad495b5b82b819121605364bbc2760dbae6e/pymdownx/highlight.py#L50-L51
2. https://github.com/facelessuser/pymdown-extensions/blob/6c3bad495b5b82b819121605364bbc2760dbae6e/pymdownx/highlight.py#L52
3. https://github.com/facelessuser/pymdown-extensions/blob/6c3bad495b5b82b819121605364bbc2760dbae6e/pymdownx/highlight.py#L194
4. https://github.com/facelessuser/pymdown-extensions/blob/29873da7921e7f2edf6c0e214ea90f501e1854b0/docs/src/markdown/extensions/highlight.md#L99

It seems that (1) is the dominating setting, and (4) documents it correctly,
but the description (2) is wrong, and the default (3) (which never really kicks in) might need to be changed to be less confusing.

**Comment by facelessuser** [comment]
Ah, yeah that is confusing. Highlight. Used to follow CodeHilite, so at some point, it used to be true but default, but when we started to move away and deviate, we started making guessing false by default because I hated language guessing. The default config controls guessing which is set to  False, though it appears I did not update the details description or internal defaults. The internal defaults never really come into play was we are always passing in the config value.

So, internally, to avoid confusion, this should be fixed.
